python/ray/worker.py

```diff
@@ -2016,8 +2026,23 @@ def connect(info,
     error_message = "Perhaps you called ray.init twice by accident?"
     assert not worker.connected, error_message
     assert worker.cached_functions_to_run is not None, error_message
+
     # Initialize some fields.
-    worker.worker_id = random_string()
+    if mode is WORKER_MODE:
+        worker.worker_id = random_string()
+    else:
+        # This is the code path of driver mode.
+        if driver_id is None:
+            driver_id = ray.ObjectID(random_string())
+
+        elif isinstance(driver_id, (bytes, str)):
```

Comment:
Let's check that it is `bytes` for Python 2 and `str` for Python 3.

Reply:
actually, for now, let's require it to be an `ObjectID`. I think the more restrictive we make it the better since it makes it easier to change later.